esp-link/cgiwifi.c

```diff
@@ -525,43 +717,50 @@ static char *wifiWarn[] = { 0,
 // print various Wifi information into json buffer
 int ICACHE_FLASH_ATTR printWifiInfo(char *buff) {
   int len;
-
-  struct station_config stconf;
-  wifi_station_get_config(&stconf);
-
-  uint8_t op = wifi_get_opmode() & 0x3;
-  char *mode = wifiMode[op];
-  char *status = "unknown";
-  int st = wifi_station_get_connect_status();
-  if (st >= 0 && st < sizeof(connStatuses)) status = connStatuses[st];
-  int p = wifi_get_phy_mode();
-  char *phy = wifiPhy[p&3];
-  char *warn = wifiWarn[op];
-  sint8 rssi = wifi_station_get_rssi();
-  if (rssi > 0) rssi = 0;
-  uint8 mac_addr[6];
-  wifi_get_macaddr(0, mac_addr);
-  uint8_t chan = wifi_get_channel();
-
-  len = os_sprintf(buff,
-    "\"mode\": \"%s\", \"modechange\": \"%s\", \"ssid\": \"%s\", \"status\": \"%s\", \"phy\": \"%s\", "
-    "\"rssi\": \"%ddB\", \"warn\": \"%s\", \"mac\":\"%02x:%02x:%02x:%02x:%02x:%02x\", \"chan\":%d",
-    mode, MODECHANGE, (char*)stconf.ssid, status, phy, rssi, warn,
-    mac_addr[0], mac_addr[1], mac_addr[2], mac_addr[3], mac_addr[4], mac_addr[5], chan);
-
-  struct ip_info info;
-  if (wifi_get_ip_info(0, &info)) {
-    len += os_sprintf(buff+len, ", \"ip\": \"%d.%d.%d.%d\"", IP2STR(&info.ip.addr));
-    len += os_sprintf(buff+len, ", \"netmask\": \"%d.%d.%d.%d\"", IP2STR(&info.netmask.addr));
-    len += os_sprintf(buff+len, ", \"gateway\": \"%d.%d.%d.%d\"", IP2STR(&info.gw.addr));
-    len += os_sprintf(buff+len, ", \"hostname\": \"%s\"", flashConfig.hostname);
-  } else {
-    len += os_sprintf(buff+len, ", \"ip\": \"-none-\"");
-  }
-  len += os_sprintf(buff+len, ", \"staticip\": \"%d.%d.%d.%d\"", IP2STR(&flashConfig.staticip));
-  len += os_sprintf(buff+len, ", \"dhcp\": \"%s\"", flashConfig.staticip > 0 ? "off" : "on");
-
-  return len;
+    //struct station_config stconf;
+    wifi_station_get_config(&stconf);
+    //struct softap_config apconf;
+    wifi_softap_get_config(&apconf);
+    
+    uint8_t op = wifi_get_opmode() & 0x3;
+    char *mode = wifiMode[op];
+    char *status = "unknown";
+    int st = wifi_station_get_connect_status();
+    if (st >= 0 && st < sizeof(connStatuses)) status = connStatuses[st];
+    int p = wifi_get_phy_mode();
+    char *phy = wifiPhy[p&3];
+    char *warn = wifiWarn[op];
+    if (op == 3) op = 4; // Done to use only one set of warnings
```

Comment:
In Soft-AP settings when in STA+AP mode, let user switch to AP mode. 
To use only one set of warnings an extra line is needed, to cover all cases.